src/roswire/ros2/reader.py

```diff
@@ -0,0 +1,132 @@
+# -*- coding: utf-8 -*-
+import json
+import os
+import shlex
+import typing
+from typing import Any, Dict, Optional, Sequence, Tuple
+
+import attr
+import pkg_resources
+from loguru import logger
+
+from ..proxy.roslaunch.config import ExecutableType, LaunchConfig, NodeConfig
+from ..proxy.roslaunch.context import LaunchContext
+from ..proxy.roslaunch.reader import LaunchFileReader
+
+if typing.TYPE_CHECKING:
+    from .. import AppInstance
+
+
+@attr.s(auto_attribs=True)
+class ROS2LaunchFileReader(LaunchFileReader):
+
+    _app_instance: 'AppInstance'
+
+    def locate_node_binary(self, package: str, node_type: str) -> str:
+        raise NotImplementedError
+
+    def read(self,
+             fn: str,
+             argv: Optional[Sequence[str]] = None
+             ) -> LaunchConfig:
+        """
+        Produces a summary of the effects of a launch file.
+
+        Parameters
+        ----------
+        fn: str
+            The name of the launch file, or an absolute path to the launch
+            file inside the container.
+        argv: Sequence[str], optional
+            An optional sequence of command-line arguments that should be
+            supplied to :code:`roslaunch`.
+
+        Raises
+        ------
+        LaunchFileNotFound
+            If the given launch file could not be found in the package.
+        """
+        if self._app_instance is None:
```

Comment:
`_app_instance` isn't optional, so the body of this if condition is unreachable.

Reply:
Fixed